CHANGES.md

```diff
@@ -1,6 +1,473 @@
+* added a compile-time error when declaring recursive records
+* improved error messages to show types of non-default numeric param values
+* made use-before-def error messages print to the symbol's definition point
+* added an error message when we can't determine an iterator's yield type
+* improved error messages for accessing tuples of types out of bounds
+* improved const checking for tuple arguments
+* added an error for passing too many arguments to a tuple-type index expression
+* added a helpful error message when throwing an error type without `new`
+* added an error message for applying `.type` to a type
+* improved checking of intents on overridden methods
+* improved the error message for library compilations using multiple modules
+* added a compile-time error for exported functions using strings
+* added an error message if `CHPL_COMM_SUBSTRATE` isn't detected by `GASNet`
+* improved error-handling for certain operations on an invalid channel
+* removed a warning regarding returned iterators being promoted to arrays
+
+Execution-time Checks
+---------------------
+* added a better error for running a `--local` program on more than 1 locale
+* added runtime memory exhaustion reporting for `CHPL_TASKS=qthreads`
+
+Bug Fixes
+---------
+* fixed a number of bugs related to initializers
+* fixed several bugs with `forwarding` fields
+* fixed several problems with calls qualified by explicit module names
+* fixed a bug in which private overloads could completely shadow public ones
+* fixed a bug in which qualified 'use's hid methods with the same name
+* fixed support for the swap operator (`<=>`) on owned and shared objects
+* fixed problems with records containing `owned` class fields
+* fixed problems with compiler-added copies for `owned` variables
+* adjusted `make install` to enable the use of the setchplenv scripts
+* fixed a bash tab completion bug for paths involving `~`
+* fixed problems with `--llvm` in `make install` builds
+* fixed problems with certain GASNet configurations using `--llvm`
+* fixed `--llvm` compiles with `CHPL_TARGET_ARCH=unknown`
+* fixed incorrect bulk-transfers between reindexed arrays
+* fixed support for `atomic real(32)` on platforms with network atomics
+* fixed bugs with operators and intents in `BigInteger` library routines
+* fixed return type of casts from `bigint` to `integral`/`real`
+* fixed bugs in `IO.readline()`
+* fixed a bug with exists(""), isFile(""), isDir(""), isLink(""), isMount("")
+* fixed problems in `FileSystem.copyFile` for files not ending in newlines
+* fixed a bug where `Search` sometimes used the wrong bounds for strided arrays
+* fixed `DynamicIterators` for 0-length ranges and domains
+* fixed a bug in mergeSort's documentation claiming it was in-place
+* fixed a bug when spawning a task with stdout set to PIPE and stderr to STDOUT
+* fixed reported line numbers for non-generic class allocations
+* fixed a broken example in the ZMQ documentation
```

Comment:
The example technically wasn't broken, just incorrectly labeled

Reply:
Are you saying this should read "fixed the label in a ZMQ example?"  Is this worth mentioning?

Reply:
I'm asking you what you want me to type here if you want this to remain; or to tell me to remove it if you think it's below the bar that anyone reading about the release would care about.  I.e., I don't want to spend any more of my effort at this point trying to figure out what this is and how to describe it.

Reply:
Okay, let's remove it